### PR TITLE
[1.12.x] Update/GenPatches for Current MCP Snapshot + Other Minor Changes

### PR DESCRIPTION
**Pull request by CDAGaming**
## Info
This PR does the following:
 - Upgrades the MCP Mappings from 20170624 to 20171009 (@mezz has already now done the same changes as in this PR, up to 20171003)
 - Fixes Deprecation Issues that would Arise if Forge moves to Gradle 3.2 or Higher
 - Typo Fixes / Code CleanUp
 - Regen Patches alongside MCP Changes

### Title has been changed to reflect upon mezz's Commit

### CODE REVIEW

**Comment by CDAGaming** [comment]
PR now updated to Reflect Changes in this Repo :)

File: SUPPORT.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ Now that you've identified the source of your issue and decided on the correct s
...
+Additionally, always check the *latest* changelog of Forge and participating mods to find out if the bug was already noticed and fixed by the developer. Often the maintainer may only support newer versions of the game, so also check the changelogs for versions of Minecraft you don't actually actively play. If the issue was adressed already, *do not report it*, even if the fix only exists for a newer version of the game, since the developer probably had a good reason to do so. If your issue only exists with versions of a mod that are for very old versions of the game, also refrain from reporting them, considering most modders only support the two latest versions of Minecraft, if at all.
```
Comment (mezz) [inline SUPPORT.md]: typo: `adressed`

File: build.gradle [hunk 2]
```diff
@@ -30,7 +30,7 @@ apply plugin: "net.minecraftforge.gradle.launch4j"
 minecraft.version = "1.12.2"
 
 minecraft {
-    mappings = 'snapshot_nodoc_20171003'
+    mappings = 'snapshot_nodoc_20171004'
```
Comment (mezz) [inline build.gradle]: Please don't change the mapping. It took me a while to test the current existing mapping and, although I doubt much has changed in one day, I don't want to have to test it again.

File: build.gradle [hunk 3]
```diff
@@ -225,20 +225,22 @@ artifacts {
...
-    def ln = "\n"; //Linux line endings because we're on git!
-
-    file.eachLine{ String s ->
-        if (s.matches("^    public static final int buildVersion    = [\\d]+;\$"))
-            s = "    public static final int buildVersion    = ${bn};";
-        if (s.matches('^    public static final String mcVersion = "[^\\"]+";'))
-            s = "    public static final String mcVersion = \"${minecraft.version}\";";
-        outfile += (s+ln);
+task ciWriteBuildNumber {
+    doLast {
```
Comment (mezz) [inline build.gradle:229]: Please explain the change from `<<` to `doLast`

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Mon Sep 14 12:28:28 PDT 2015
+#Tue Oct 03 04:41:15 CDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
```
Comment (mezz) [inline gradle/wrapper/gradle-wrapper.properties]: Please do not change the gradle distribution unless there is a reason.

File: gradlew [hunk 5]
```diff
@@ -1,25 +1,43 @@
...
 
 ##############################################################################
 ##
 ##  Gradle start up script for UN*X
 ##
 ##############################################################################
 
-# Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+# Attempt to set APP_HOME
```
Comment (mezz) [inline gradlew]: Please do not change the gradle distribution unless there is a reason.

File: gradlew.bat [hunk 6]
```diff
@@ -8,14 +8,14 @@
...
 
-@rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-set DEFAULT_JVM_OPTS=
-
 set DIRNAME=%~dp0
 if "%DIRNAME%" == "" set DIRNAME=.
 set APP_BASE_NAME=%~n0
 set APP_HOME=%DIRNAME%
 
+@rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
```
Comment (mezz) [inline gradlew.bat:16]: Please do not change the gradle distribution unless there is a reason.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by CDAGaming** [comment]
Review Comments
A. As for the Mappings, keep in mind the same changes you applied to update the Mappings on the official repo were already on my fork a few days before (I simply didn't send a PR at that time just so I can be sure everything worked properly). Each time I had updated the mappings, I compiled and then used that Forge version to make sure everything works as it should the same as in the previous mappings.

B. As for doLast and >>, In Gradle 3.2 and Later this change was because of a deprecation warning on >> and merely this is just replacing it with doLast to solve that warning (In the warning, it also states to use doLast instead)

C. The Gradle Distribution Files were automatically changed (A Command, which I'll paste here later, was used for the upgrade)

File: SUPPORT.md (hunk 1 above)
Comment (CDAGaming) [reply SUPPORT.md]: Fixed, and Commiting shortly, don't know how my mind skipped over that one 👍

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -15,7 +15,7 @@ Contributor License Agreement
 - You grant Forge a license to use your code contributed to the primary codebase (everything **not** under patches) in Forge, under the LGPLv2.1 license.
 - You assign copyright ownership of your contributions to the patches codebase (everything under patches) to Forge, where it will be licensed under the LGPLv2.1 license.
 
-Additionally, while 1.9.4 is being maintained, you will be asked to acknowledge these two additional clauses.
+Additionally, while 1.12.2 is being maintained, you will be asked to acknowledge these two additional clauses.
```
Comment (tomelfring) [inline CONTRIBUTING.md]: I don't think this is an typo

Comment (CDAGaming) [reply CONTRIBUTING.md]: It isn't a typo, I just changed it to 1.12.2 as it is the Latest Maintained Version. @thommy101

Comment (williewillus) [reply CONTRIBUTING.md]: No, they mean that this changes the meaning of the document. The additional two clauses are for the old license which does not apply after 1.10

Comment (CDAGaming) [reply CONTRIBUTING.md]: Ah, Ok. In that case, will fix this shortly (MCP Tests were also working so the mappings will be on 20171005 in the next commit

Comment (CDAGaming) [reply CONTRIBUTING.md]: Fixed!

**Comment by mezz** [comment]
What is your reason for updating Gradle from 2.14-bin to 3.5-all? Is there a need for it?

**Comment by kashike** [comment]
Regarding the Gradle version bump: bug fixes, performance improvements, etc.

**Comment by CDAGaming** [comment]
As kashike said, that's mostly the reason why Gradle was upgraded (Never tested Gradle 4, as i do not think anything related to Forge has gotten to Gradle 4 just yet, with the exception of a small amount of mods), as well as the `doLast` change as `>>` (leftshift) would be deprecated in Gradle 5.0, if Forge ever goes there in the foreseeable future, so `doLast` was merely to future-proof that area.

**Comment by mezz** [comment]
I haven't experienced any bugs or performance problems with the current version.
On most of my projects I'm using Gradle 4.0 and that seems to work fine too.

Where did 3.5 come from? It seems arbitrary and your responses are not exactly filling me with confidence in your decision...

Please leave the gradle versions alone for now. If a pressing reason to upgrade comes up we can do it then, but right now I don't really want to be responsible for accidentally breaking some unknown and arcane gradle thing in Forge's workflow (again).

**Comment by CDAGaming** [comment]
Ok then, in that case we come up on the next decision, once I downgrade Gradle, should I still keep the deprecation prevention change from >> to doLast or Revert that as well for a later date if/when we go back to Gradle 3.x?

(MCP 20171006 is also now tested to be Ok, so that'll also be with the next commit in this PR)

**Comment by mezz** [comment]
Changing to `doLast` is fine, assuming this gradle version supports it.

**Comment by CDAGaming** [comment]
It should, i'll send the commit in tomorrow morning after i test it

File: build.gradle (hunk 3 above)
Comment (liach) [reply build.gradle:229]: to solve gradle warnings

Comment (CDAGaming) [reply build.gradle:229]: It solves deprecation warnings on Gradle 3.2 and Up, should Forge ever officially change to 3.2+

**Comment by CDAGaming** [comment]
Well, i have no clue why it didn't solve the conflicts the first time, but it should be solved

(Though i still gotta recheck this PR later to be sure)

**Comment by CDAGaming** [comment]
Oh no you don't. Not dead yet...lemme rebase this later.

**Comment by mezz** [comment]
I'm sorry but I don't think we are on the same page here.
If you want to fix typos across all the code, please do.
However you should not change the gradle version or update mcp.

Updating gradle has no real benefits to this project right now and only opens up possible issues because of the complexity of the forge gradle system used to put this project together.
Updating mcp often breaks other PRs and forces them to update, so we only update mcp when it's really necessary or there is a good opportunity to.

**Comment by CDAGaming** [comment]
Going to Temporarily Close this, and will reopen as a New PR (Will be able to better maintain the new PR)

**Pull request closed without merging**